parse/go.mod

```diff
@@ -0,0 +1,22 @@
+module github.com/kwilteam/kwil-db/parse
+
+go 1.21.0
+
+replace github.com/kwilteam/kwil-db/core => ../core
+
+require (
+	github.com/antlr4-go/antlr/v4 v4.13.0
+	github.com/kwilteam/kwil-db v0.7.2
```

Comment:
This almost defeats the whole purpose of making a parse module.  `parse` consumers don't want to require the main module, and we don't want a circular `parse` <-> main module require.

Reply:
I see it used in parse/sql/sqlanalyzer/clean/clean.go for `validation.IsKeyword`.

I guess you wanted to use `core/types/validation.IsKeyword` instead?

I also see the only other users of `validation.IsKeyword` are `validation.ValidateIdentifier` and `validation.ValidateDBID`, and both of those seem to be unused?  (there's a also now a `core/types/validation.ValidateIdentifier` that seems to provide this now, so I bet the common/validation is now obsolete?

Reply:
Ah, yeah, that's all it was.

`import "github.com/kwilteam/kwil-db/core/types/validation"

in parse/sql/sqlanalyzer/clean/clean.go

seems to remove this require.

Reply:
Oh wow total mistake on my end. That old validation package shouldn't be there, definitely meant to use core

Reply:
TIL you can have circular dependencies between submodules